qiskit/transpiler/__init__.py

```diff
@@ -40,629 +41,787 @@
-       from qiskit.providers.fake_provider import FakeLagosV2
-
-       backend = FakeLagosV2()
-       pass_manager = generate_preset_pass_manager(3, backend)
-
-   which will generate a :class:`~.StagedPassManager` object for optimization level 3
-   targeting the :class:`~.FakeLagosV2` backend (equivalent to what is used internally
-   by :func:`~.transpile` with ``backend=FakeLagosV2()`` and ``optimization_level=3``).
-   You can use this just like working with any other :class:`~.PassManager`. However,
-   because it is a :class:`~.StagedPassManager` it also makes it easy to compose and/or
-   replace stages of the pipeline. For example, if you wanted to run a custom scheduling
-   stage using dynamical decoupling (via the :class:`~.PadDynamicalDecoupling` pass) and
-   also add initial logical optimization prior to routing you would do something like
-   (building off the previous example):
+.. jupyter-execute::
 
-   .. code-block:: python
+   from qiskit.visualization.timeline import draw as timeline_draw
 
-       from qiskit.circuit.library import XGate, HGate, RXGate, PhaseGate, TGate, TdgGate
-       from qiskit.transpiler import PassManager
-       from qiskit.transpiler.passes import ALAPScheduleAnalysis, PadDynamicalDecoupling
-       from qiskit.transpiler.passes import CXCancellation, InverseCancellation
+   timeline_draw(circ)
+
+The scheduling of a circuit involves two parts, analysis and constraint mapping followed by a
+padding pass. The first part requires running a scheduling analysis pass such as
+:class:`~.ALAPSchedulingAnalysis` or :class:`~.ASAPSchedulingAnalysis` which analyzes the circuit
+and records the start time of each instruction in the circuit using a scheduling algorithm ("as late
+as possible" for  :class:`~.ALAPSchedulingAnalysis` and "as soon as possible" for
+:class:`~.ASAPSchedulingAnalysis`) in the property set. Once the circuit has an initial scheduling
+additional passes can be run to account for any timing constraints on the target backend, such
+as alignment constraints. This is typically done with the
+:class:`~.ConstrainedReschedule` pass which will adjust the scheduling
+set in the property set to the constraints of the target backend. Once all
+the scheduling and adjustments/rescheduling are finished a padding pass,
+such as :class:`~.PadDelay` or :class:`~.PadDynamicalDecoupling` is run
+to insert the instructions into the circuit, which completes the scheduling.
 
-       backend_durations = backend.target.durations()
-       dd_sequence = [XGate(), XGate()]
-       scheduling_pm = PassManager([
-           ALAPScheduleAnalysis(backend_durations),
-           PadDynamicalDecoupling(backend_durations, dd_sequence),
-       ])
-       inverse_gate_list = [
-           HGate(),
-           (RXGate(np.pi / 4), RXGate(-np.pi / 4)),
-           (PhaseGate(np.pi / 4), PhaseGate(-np.pi / 4)),
-           (TGate(), TdgGate()),
+Scheduling Analysis with control flow instructions:
```

Comment:
This line looks like it might want to be a separate section heading.  If we don't want to make a true heading, it could just be a
```rst
.. rubric:: Scheduling analysis with control-flow instructions
```

That said, the whole sectioning of this part is a bit funky.

Reply:
Yeah this was a limitation with the previous structure where it was done via a drop down. It was originally written as a docstring for a base class which we didn't publish docs with a proper heading hierarchy and we refactored it to work in the old structure. I'll update this to have proper headings